flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/batch/compact/BatchPartitionCommitOutputFormat.java

```diff
@@ -0,0 +1,148 @@
+        implements OutputFormat<CompactOutput>, FinalizeOnMaster, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private final FileSystemFactory fsFactory;
+    private final TableMetaStoreFactory msFactory;
+    private final PartitionCommitPolicyFactory partitionCommitPolicyFactory;
+    private final Path tmpPath;
+    private final boolean overwrite;
+    private final boolean isToLocal;
+    private final String[] partitionColumns;
+    private final LinkedHashMap<String, String> staticPartitions;
+    private final ObjectIdentifier identifier;
+
+    private transient Map<String, List<Path>> partitionsFiles;
+
+    public BatchPartitionCommitOutputFormat(
+            FileSystemFactory fsFactory,
+            TableMetaStoreFactory msFactory,
+            boolean overwrite,
+            boolean isToLocal,
+            Path tmpPath,
+            String[] partitionColumns,
+            LinkedHashMap<String, String> staticPartitions,
+            ObjectIdentifier identifier,
+            PartitionCommitPolicyFactory partitionCommitPolicyFactory) {
+        this.fsFactory = fsFactory;
+        this.msFactory = msFactory;
+        this.partitionCommitPolicyFactory = partitionCommitPolicyFactory;
+        this.tmpPath = tmpPath;
+        this.identifier = identifier;
+        this.overwrite = overwrite;
+        this.isToLocal = isToLocal;
+        this.partitionColumns = partitionColumns;
+        this.staticPartitions = staticPartitions;
+    }
+
+    @Override
+    public void finalizeGlobal(int parallelism) throws IOException {
+        try {
+            List<PartitionCommitPolicy> policies = Collections.emptyList();
+            if (partitionCommitPolicyFactory != null) {
+                policies =
+                        partitionCommitPolicyFactory.createPolicyChain(
+                                Thread.currentThread().getContextClassLoader(),
+                                () -> {
+                                    try {
+                                        return fsFactory.create(tmpPath.toUri());
+                                    } catch (IOException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                });
+            }
+
+            // commit the partitions with the given files
+            // it will move the written temporary files to partition's location
+            FileSystemCommitter committer =
+                    new FileSystemCommitter(
+                            fsFactory,
+                            msFactory,
+                            overwrite,
+                            tmpPath,
+                            partitionColumns.length,
+                            isToLocal,
+                            identifier,
+                            staticPartitions,
+                            policies);
+            committer.commitPartitionsWithFiles(partitionsFiles);
```

Comment:
I'm not sure whether we could get right value of `partitionsFiles` when run `finalizeGlobal` in JobMaster.

Reply:
Yes, you're right. Now, i replace it with `BatchPartitionCommitterSink`.